server/go/websocket/router.go

```diff
@@ -0,0 +1,47 @@
+package websocket
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/gorilla/websocket"
+)
+
+type MessageExecutor func(*websocket.Conn, Message)
+
+type Router struct {
+	routes map[string]MessageExecutor
+}
+
+func NewRouter() *Router {
+	return &Router{
+		routes: make(map[string]MessageExecutor),
```

Comment:
Another noob one. What are the route names/message types that would be available here?

Reply:
What that code means is that you will have a way to map a message type to an "object" that knows how to handle that specifc type